static/app/views/insights/common/views/spans/selectors/domainSelector.tsx

```diff
@@ -87,24 +89,39 @@ export function DomainSelector({
   });
 
   const incomingDomains = [
-    ...uniq(domainData?.flatMap(row => row[SpanMetricsField.SPAN_DOMAIN])),
+    ...uniq(
+      domainData?.flatMap(row => {
+        const spanDomain: string | string[] = row[SpanMetricsField.SPAN_DOMAIN];
+        // `useInsightsEap` returns a string, but `metrics` returns an array
+        if (typeof spanDomain === 'string') {
+          return spanDomain.split(',');
+        }
+        return spanDomain;
+      })
+    ).filter(Boolean),
   ];
 
+  const domainList = incomingDomains.map(domain => ({
+    label: domain,
+    value: useEap ? `*${domain}*` : domain,
```

Comment:
I think this should be:
```suggestion
    value: useEap ? `*,${domain},*` : domain,
```

🤔 If there's a table called `app_users` and another table called `app_usersaccounts`, without commas, selecting `app_users` in the dropdown would search for `*app_users*` and return data from both tables. We need to search for `*,app_users,*` to make sure the filtering works correctly.

Reply:
Hmm, I don't think this will work for resource module, where span.domain is a single value containing no commas. Will confirm tmrw!

But regardless, I think I can update to work like this when there are commas